src/SFA.DAS.AANHub.Application.UnitTests/Apprentices/Commands/PatchApprenticeMember/PatchApprenticeMemberCommandValidatorTests.cs

```diff
@@ -0,0 +1,161 @@
+
+        public PatchApprenticeMemberCommandValidatorTests()
+        {
+            _memberReadRepository = new Mock<IMembersReadRepository>();
+        }
+
+        private const string Name = "Name";
+        private const string Email = "Email";
+
+        [TestCase("replace", true)]
+        [TestCase("remove", false)]
+        public async Task Validate_Patch_Name_NoMatchingOperation_ErrorMessage(string operation, bool isValid)
+        {
+            var memberId = Guid.NewGuid();
+            var apprenticeid = 123;
+            var nameValue = "test";
+
+            var command = new PatchApprenticeMemberCommand
+            {
+                RequestedByMemberId = memberId,
+                ApprenticeId = apprenticeid,
+                Patchdoc = new JsonPatchDocument<Apprentice>()
+            };
+
+            command.Patchdoc = new JsonPatchDocument<Apprentice>
+            {
+                Operations =
+                {
+                    new Operation<Apprentice>
+                        { op = operation, path = Name, value = nameValue }
+                }
+            };
+
+            var member = new Member() { Status = Domain.Common.Constants.MembershipStatus.Live };
+            _memberReadRepository.Setup(a => a.GetMember(memberId)).ReturnsAsync(member);
+
+            var sut = new PatchApprenticeMemberCommandValidator(_memberReadRepository.Object);
+
+            var result = await sut.TestValidateAsync(command);
+
+            if (isValid)
+                result.ShouldNotHaveValidationErrorFor(p => p.Patchdoc.Operations.Count(c => c.op.Equals(operation)));
+            else
+                result.ShouldHaveValidationErrorFor(p => p.Patchdoc.Operations.Count(c => c.op.Equals(operation)));
+        }
+
+        [Test]
+        public async Task Validate_Patch_NoOperations_ErrorMessage()
+        {
+            var memberId = Guid.NewGuid();
+            var apprenticeid = 123;
+
+            var command = new PatchApprenticeMemberCommand
+            {
+                RequestedByMemberId = memberId,
+                ApprenticeId = apprenticeid,
+                Patchdoc = new JsonPatchDocument<Apprentice>()
+            };
+
+            var member = new Member() { Status = Domain.Common.Constants.MembershipStatus.Live };
+            _memberReadRepository.Setup(a => a.GetMember(memberId)).ReturnsAsync(member);
+
+            var sut = new PatchApprenticeMemberCommandValidator(_memberReadRepository.Object);
+
+            var result = await sut.TestValidateAsync(command);
+
+            result.ShouldHaveValidationErrorFor(c => c.Patchdoc.Operations.Count);
+            Assert.IsFalse(result.IsValid);
+            Assert.IsTrue(result.Errors.Count > 0);
+            Assert.AreEqual(PatchApprenticeMemberCommandValidator.NoPatchOperationsPresentErrorMessage, result.Errors[0].ErrorMessage);
+        }
+
+        [Test]
+        public async Task Validate_Patch_Name_OperationsExceeded_ErrorMessage()
```

Comment:
ValidatePatchDoc_DuplicateReplaceNameOperation_InvalidResponse

Reply:
Is now ValidatePatchDoc_DuplicateReplaceOperation_InvalidResponse. Doesn't have Name in between - DuplicateNameReplace as it is now common for both name and email